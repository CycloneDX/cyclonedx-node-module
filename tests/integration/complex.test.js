/* eslint-env jest */

/*
 * This file is part of CycloneDX Node Module.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * Copyright (c) OWASP Foundation. All Rights Reserved.
 */

const Bom = require('../../model/Bom')
const Tool = require('../../model/Tool')
const Metadata = require('../../model/Metadata')
const Dependency = require('../../model/Dependency')
const Component = require('../../model/Component')

describe('integration of complex', () => {
  const complex = new Bom()
  complex.metadata = new Metadata()
  complex.metadata.timestamp = new Date(0)
  complex.metadata.tools.push(new Tool('foo', 'bar', '0.0.0-testing'))
  complex.metadata.component = new Component()
  complex.metadata.component.name = 'RootComponent'
  complex.metadata.component.version = '1.33.7'
  complex.metadata.component.bomRef = 'myRootComponent'
  const c1 = new Component()
  c1.name = 'SomeComponent'
  c1.version = '23.42'
  c1.bomRef = 'some-component-2342'
  complex.components.push(c1)
  const c2 = new Component()
  c2.name = 'SomeComponent'
  c2.version = '0.8.15'
  c2.bomRef = 'some-component-0815'
  complex.components.push(c2)
  complex.addDependency(new Dependency(complex.metadata.component.bomRef, [
    new Dependency(c1.bomRef),
    new Dependency(c2.bomRef)
  ]))

  test.each(
    [
      'XML',
      'JSON'
    ]
  )('as %s', (target, done) => {
    process.env.BOM_REPRODUCIBLE = '1'
    const result = complex[`to${target}`]()
    expect(result).toMatchSnapshot()
    done()
  })
})
