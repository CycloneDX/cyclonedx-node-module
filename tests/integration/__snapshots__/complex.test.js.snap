// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration of complex as JSON 1`] = `
"{
  \\"bomFormat\\": \\"CycloneDX\\",
  \\"specVersion\\": \\"1.3\\",
  \\"version\\": 1,
  \\"metadata\\": {
    \\"tools\\": [
      {
        \\"vendor\\": \\"foo\\",
        \\"name\\": \\"bar\\",
        \\"version\\": \\"0.0.0-testing\\"
      }
    ],
    \\"component\\": {
      \\"type\\": \\"library\\",
      \\"bom-ref\\": \\"myRootComponent\\",
      \\"name\\": \\"RootComponent\\",
      \\"version\\": \\"1.33.7\\"
    }
  },
  \\"components\\": [
    {
      \\"type\\": \\"library\\",
      \\"bom-ref\\": \\"some-component-0815\\",
      \\"name\\": \\"SomeComponent\\",
      \\"version\\": \\"0.8.15\\"
    },
    {
      \\"type\\": \\"library\\",
      \\"bom-ref\\": \\"some-component-2342\\",
      \\"name\\": \\"SomeComponent\\",
      \\"version\\": \\"23.42\\"
    }
  ],
  \\"dependencies\\": [
    {
      \\"ref\\": \\"myRootComponent\\",
      \\"dependsOn\\": [
        \\"some-component-0815\\",
        \\"some-component-2342\\"
      ]
    }
  ]
}"
`;

exports[`integration of complex as XML 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<bom xmlns=\\"http://cyclonedx.org/schema/bom/1.3\\" version=\\"1\\">
  <metadata>
    <tools>
      <tool>
        <vendor>foo</vendor>
        <name>bar</name>
        <version>0.0.0-testing</version>
      </tool>
    </tools>
    <component type=\\"library\\" bom-ref=\\"myRootComponent\\">
      <name>RootComponent</name>
      <version>1.33.7</version>
    </component>
  </metadata>
  <components>
    <component type=\\"library\\" bom-ref=\\"some-component-0815\\">
      <name>SomeComponent</name>
      <version>0.8.15</version>
    </component>
    <component type=\\"library\\" bom-ref=\\"some-component-2342\\">
      <name>SomeComponent</name>
      <version>23.42</version>
    </component>
  </components>
  <dependencies>
    <dependency ref=\\"myRootComponent\\">
      <dependency ref=\\"some-component-0815\\"/>
      <dependency ref=\\"some-component-2342\\"/>
    </dependency>
  </dependencies>
</bom>"
`;
